Added Nord based theme

joelchrono12 opened the pull request:
Closes #

Changes proposed in this pull request:

- A new theme!-

How to test the feature manually:

1. Pick it from the settings in Display

I am still kinda noob at this so if anything else is needed please let me know

joelchrono12 committed:
added new theme

joelchrono12 committed:
removed old picture

joelchrono12 committed:
modified color of summary and author

joelchrono12 committed:
fixed input elements being hard to distinguish

Alkarex committed:
make fix-all

Alkarex committed:
Merge branch 'edge' into edge

Alkarex commented:
Hello and thanks for your contribution 👍🏻 
I have automatically fixed a number of issues by running `make fix-all` and `make test-all`, but there are a few issues remaining (see the output of the automated tests below)

Alkarex committed:
markdown lint

joelchrono12 commented:
> Hello and thanks for your contribution 👍🏻 I have automatically fixed a number of issues by running `make fix-all` and `make test-all`, but there are a few issues remaining (see the output of the automated tests below)

Thank you, do you know of how to pull the changes you did to my repo, so I can fix these problems?

Alkarex commented:
> Thank you, do you know of how to pull the changes you did to my repo, so I can fix these problems?

```sh
git pull --ff-only
```

joelchrono12 committed:
Merge branch 'edge' of github:joelchrono12/FreshRSS into edge

joelchrono12 committed:
cleaned up css, still incomplete afaik

joelchrono12 committed:
removed duplicate, highlight current article

joelchrono12 committed:
removed more duplicates, reordered some things

math-GH commented:
regarding your CSS:

Here you can find the defined order of CSS properties:

https://github.com/FreshRSS/FreshRSS/blob/f85c510ed49be031145f6b35e815ce890cd4f9aa/.stylelintrc.json#L40-L65

Alkarex committed:
make fix-all

Alkarex commented:
For the order of attributes and other cosmetic things, running `make fix-all` automatically fixes all that. I have just done it, so the few remaining errors are those that must be manually handled

joelchrono12 commented:
> regarding your CSS:
> 
> Here you can find the defined order of CSS properties:
> 
> https://github.com/FreshRSS/FreshRSS/blob/f85c510ed49be031145f6b35e815ce890cd4f9aa/.stylelintrc.json#L40-L65

Thanks, I noticed there are some tests which let me know the line numbers where the css is wrong. But as I edit changes those line numbers wrong and I kinda laze around.

Is there a way to run those tests locally? I think I lack some dependencies but idk which ones, sorry if this is in the documentation, I should re-check

joelchrono12 commented:
Well, not a lot of errors left, I'll fixed them now

Alkarex commented:
To run tests locally, try `make fix-all` and `make test-all`
Otherwise in your case, `npm install` and `npm run fix` and `npm test` will cover most of what you need.
In both cases, nodejs + npm must to be available

joelchrono12 committed:
removed last few css duplicates

joelchrono12 commented:
Well now there are more errors 🙃

joelchrono12 committed:
removed empty css blocks

Alkarex committed:
make fix-all

joelchrono12 commented:
Hey, it passed, thank you @Alkarex I couldn't manage to get the make running since I still lacked some smaller depenencies. But now its working!

math-GH commented:
Please check the menus in mobile view.

The "user query" submenu is broken (the white rhombus is part of the submenu)

![grafik](https://user-images.githubusercontent.com/1645099/172672069-ec091317-b048-4fb0-8c98-a78d519034d8.png)

And the config menu: the close button looks not styled:
![grafik](https://user-images.githubusercontent.com/1645099/172672229-b7507ccb-ef97-43d2-83b0-7b4a86b57a56.png)

math-GH committed:
Cleaned CSS - deleted styles that are maintained in template.css

math-GH commented:
> Please check the menus in mobile view.

I cleaned your CSS and fixed that issue. 

Now it is much more simpler to maintain the CSS code via template.css

math-GH committed:
RTL CSS

joelchrono12 committed:
color changed from template

joelchrono12 commented:
> > Please check the menus in mobile view.
> 
> I cleaned your CSS and fixed that issue.
> 
> Now it is much more simpler to maintain the CSS code via template.css

I see, thank you, sadly the color from the template is white so that little square still looked wrong, I just fixed it

math-GH committed:
icons deleted that did not change to the default icons

math-GH commented:
There is a `import.svg` file in your theme. Where do you use it? It is not part of the default icon set.

math-GH committed:
Fix CSS (tabs instead of spaces) + RTL

joelchrono12 commented:
> There is a `import.svg` file in your theme. Where do you use it? It is not part of the default icon set.

Probably nowhere, or at least I don't remember...
By the way, I just noticed my branch is still the "edge" one in my fork.

I should have created a new branch where these commits are done, right? is there a way to fix this myself or can any of you work around it?

math-GH commented:
> By the way, I just noticed my branch is still the "edge" one in my fork.
> 
> I should have created a new branch where these commits are done, right? is there a way to fix this myself or can any of you work around it?

you do not need a new branch. Your `edge` branch is used for this PR. To update your branch you could open https://github.com/joelchrono12/FreshRSS and click on the `fetch upstream` button (not visible in the screen shot) 
![grafik](https://user-images.githubusercontent.com/1645099/172923244-07dbc813-2b93-4c4f-94f0-363fa7dfc761.png)
click then on `fetch and merge`
![grafik](https://user-images.githubusercontent.com/1645099/172923445-d4c86d44-ba5b-4ce9-8fa2-6b404a114522.png)

math-GH committed:
Delete import.svg

math-GH commented:
I deleted `import.svg` (The Swage theme uses import.svg too, but it is unused there too)

math-GH committed:
use CSS filter instead of coloured theme icons

math-GH commented:
With this changes in [6e08ead](https://github.com/FreshRSS/FreshRSS/pull/4400/commits/6e08ead20bb56c4e6a1c104e89e09dc9b8d78415) the icons are now coloured with CSS filters. It improves the maintenance a lot.

math-GH commented:
@joelchrono12 Could you please add a style, that highlights active buttons?
These buttons looks the same when they are active and inactive
![grafik](https://user-images.githubusercontent.com/1645099/172939905-ae1dac6d-0151-49be-91af-d2fefed2bfe0.png)

math-GH commented on p/themes/Fresh-Nord/nord.css:
```diff
@@ -0,0 +1,1228 @@
+
+/*=== Links */
+a {
+	background: none !important;
```
is this `!important` really needed?

joelchrono12 replied:
> is this `!important` really needed?

Its not now, since I added a enabled a border to distinguish active buttons... I actually deleted the line completely, seems fine.

![image](https://user-images.githubusercontent.com/18448587/172953001-4a0ce0da-be87-4aff-a713-f8e93dce70d3.png)

joelchrono12 committed:
active buttons are noticeable, more colored icons

joelchrono12 committed:
removed unnecesary line

joelchrono12 committed:
added background to favicons

math-GH commented on p/themes/Fresh-Nord/nord.css:
```diff
@@ -318,7 +319,7 @@ a.btn {
 /*=== Icons */
 .icon {
-	filter: invert(74%) sepia(29%) saturate(411%) hue-rotate(171deg) brightness(130%) contrast(85%);
+	/*filter: invert(74%) sepia(29%) saturate(411%) hue-rotate(171deg) brightness(130%) contrast(85%);*/
 }
```
Is it on purpose?

joelchrono12 replied:
Yeah, because the icons stopped being the right color. I was going to mention that in my previous comment but I forgot...
I was considering using that for "active" buttons so the svg would be the same color as the background, but I couldn't figure it out, so the border will do :P

I would have removed .icon altogether but I wasn't sure

joelchrono12 replied:
To explain a bit more. I originally intended for active buttons to have the same light blue color, and the svg files be the background color via CSS and filter, but I didn't want to bother tbh

joelchrono12 committed:
Merge branch 'edge' into edge

Alkarex committed:
Merge branch 'edge' into edge

Alkarex committed:
make fix-all

Alkarex committed:
RTL

joelchrono12 committed:
removed background color change since its not present in other menus

Alkarex committed:
make fix-all

Alkarex commented:
Please try to run `make fix-all` and `make test-all` as it will automatically fix most trivial errors and catch the rest

Alkarex commented:
I have just tried this PR and the theme looks very nice, congrats 👍🏻 

One thing I have spotted is that `.alert-success` is not very distinct from a normal text:
![image](https://user-images.githubusercontent.com/1008324/174111669-5c273868-d76b-496d-bb78-c2935a3de6e8.png)

Compare with e.g. *Origine*:

![image](https://user-images.githubusercontent.com/1008324/174112157-504bdfab-2e7a-42ef-8bc6-e86206d657da.png)

Alkarex commented on p/themes/Fresh-Nord/metadata.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Nord theme",
```
Could you please use the same name than the directory? It has an impact on the alphabetic order in the gallery of themes.
So either *Fresh-Nord...* here, or change the name  (`git mv`) of the folder to *Nord...*

joelchrono12 replied:
Done, thanks for the reminder!

Alkarex commented on p/themes/Fresh-Nord/nord.css:
```diff
@@ -0,0 +1,1232 @@
+html, body {
+	background: var(--bg);
+	color: var(--text);
+	font-family: var(--sans-font) !important;
```
If some of the remaining `!important` are not required, please remove them

joelchrono12 replied:
I did remove some of them. Mostly the ones I added myself

joelchrono12 committed:
Merge branch 'edge' into edge

joelchrono12 committed:
fixed alerts, code sections

joelchrono12 committed:
fixed some css being inconsistent

joelchrono12 committed:
renamed to Nord

joelchrono12 committed:
removed important from css

joelchrono12 commented:
I still can't run make test-all and such, I have no idea of which dependencies I am missing to be honest :c

math-GH committed:
Reorder CSS, fix duplicate, update RTL CSS

math-GH commented:
The contrast of "Please add some feed" is very low to the yellow message box

![grafik](https://user-images.githubusercontent.com/1645099/174433657-e4143009-ae52-4351-a028-d7c175d736ad.png)

Alkarex commented:
> I still can't run make test-all and such, I have no idea of which dependencies I am missing to be honest :c

@joelchrono12 Let's try to fix that. First, on what platform / OS do your run? What is the error you get?

## Debian / Ubuntu

### Full tests

```sh
sudo apt-get update && sudo apt-get install --no-install-recommends -y \
	make php-cli php-mbstring php-xml npm

make fix-all
make test-all
```

### Only npm-based tests

```sh
sudo apt-get update && sudo apt-get install --no-install-recommends -y \
	npm

npm install
npm run fix 
npm test
```

joelchrono12 committed:
ran npm fix and test. Removed duplicates

joelchrono12 committed:
Merge branch 'edge' of github:joelchrono12/FreshRSS into edge

joelchrono12 commented:
Oh this is kinda cool, it reorders stuff for me, it was a pain to do myself xD, using make did not work, it showed me this after installed what you suggested. I am running Fedora btw.

![image](https://user-images.githubusercontent.com/18448587/174442892-63c34527-2bc2-4af1-b64c-a239f99adea0.png)

The npm tests worked fine tho

joelchrono12 committed:
fixed link color in warnings

joelchrono12 committed:
changed alert-warn colors

Alkarex commented:
> Oh this is kinda cool, it reorders stuff for me, it was a pain to do myself xD, using make did not work, it showed me this after installed what you suggested. I am running Fedora btw.

@joelchrono12 On Fedora, please try the following:

```sh
yum install git make npm php-cli php-mbstring php-xml php-pdo unzip wget
```

joelchrono12 commented:
> > Oh this is kinda cool, it reorders stuff for me, it was a pain to do myself xD, using make did not work, it showed me this after installed what you suggested. I am running Fedora btw.
> 
> @joelchrono12 On Fedora, please try the following:
> 
> ```shell
> yum install git make npm php-cli php-mbstring php-xml php-pdo unzip wget
> ```

Thank you, this worked, the phpstan error you fixed on your commit showed up but I assume it has nothing to do with css themes lol

joelchrono12 commented:
Also I just noticed that in your screenshot the icons are not colored as I initially intended. Here is how they look in my case. The color is still hardcoded on the svg files...

![image](https://user-images.githubusercontent.com/18448587/174451174-2750e7ef-523c-49f0-8f67-53d9b5a5037f.png)

joelchrono12 committed:
color of icons can be changed with css

joelchrono12 commented:
Now I understand why you added the "filter" thing to the .icon class. I just reverted that so the icons are the right color

Alkarex commented:
Hint: When there are automatic fixes applied, you might have to run `make fix-all` once more, for instance when some types of fixes in the normal CSS files needs to also be applied on the RTL version, as is the case now.

Alkarex committed:
make fix-all

joelchrono12 committed:
oops, fixed spaces instead of tabs

joelchrono12 committed:
Merge branch 'edge' of github:joelchrono12/FreshRSS into edge

joelchrono12 commented:
> Hint: When there are automatic fixes applied, you might have to run `make fix-all` once more, for instance when some types of fixes in the normal CSS files needs to also be applied on the RTL version, as is the case now.

oh great I did that at the same time as you xD

Alkarex commented:
Hint 2: We are using https://editorconfig.org/ to specify whitespace policies for various types of files in a standard way. This is supported by most editors, either natively or thanks to a plugin

joelchrono12 commented:
> Hint 2: We are using https://editorconfig.org/ to specify whitespace policies for various types of files in a standard way. This is supported by most editors, either natively or thanks to a plugin

Thank you, I installed the plugin for neovim and also removed some configs I had set myself that were conflicting. It should be good now! Sorry

Alkarex committed:
Merge branch 'edge' into edge

Alkarex commented:
Please double-check the *global view*

![image](https://user-images.githubusercontent.com/1008324/174893447-438fc1bf-4789-49f6-b203-a30ff485061a.png)

A few things are strange, in particular the background colour of article lists:

![image](https://user-images.githubusercontent.com/1008324/174893239-55d8dedd-e5b8-4efc-b226-ec8b566a137c.png)

joelchrono12 commented:
I cannot replicate this issue, I tried both in chromium and firefox...

joelchrono12 committed:
border color fix

joelchrono12 committed:
border color fix

joelchrono12 committed:
Merge branch 'edge' of github:joelchrono12/FreshRSS into edge

math-GH commented:
> Please double-check the _global view_
> 
> 
> A few things are strange, in particular the background colour of article lists:
> 

I can confirm this issue with Firefox on Windows

math-GH committed:
Merge branch 'edge' into edge

joelchrono12 commented:
> I can confirm this issue with Firefox on Windows

I don't really know how to solve this, my guess is that some  some CSS needs an `!important` added to it. If any of you could find which one is it and add it it would be better since I can't get this to work...

math-GH committed:
Global view: dark background

math-GH commented:
> > I can confirm this issue with Firefox on Windows
> 
> I don't really know how to solve this, my guess is that some some CSS needs an `!important` added to it. If any of you could find which one is it and add it it would be better since I can't get this to work...

fixed

joelchrono12 committed:
Merge branch 'edge' into edge

Alkarex commented:
Thanks again!
Please add a line for you in https://github.com/FreshRSS/FreshRSS/blob/edge/CREDITS.md

Alwaysin commented:
I like the theme! :+1:

There seems to not be the new message notification banner, is this a feature?

joelchrono12 commented:
> I like the theme! 👍
> 
> There seems to not be the new message notification banner, is this a feature?

This is based on the default template and it has nothing else added.
When there are new posts available there is a little message showing up a the top of the feed, do you mean that?

Alwaysin commented:
Yes, I meant that. Sorry, I probably had a problem on my side, it is now showing correctly!

Alwaysin commented:
I think there's definetly a problem with this theme. On latest Chrome version on Windows, my tab of FreshRSS uses less than 200Mo of memory. When I switch to this theme, it uses circa 1Go. That's probably why it sometimes don't show the banner.
It is even more visible on a mobile phone, everything is so slow.

EDIT: Got it, the theme is loading EVERY article. I have 14000+ unread, so yeah.

math-GH commented:
The theme has no effect on the loading behavior. The theme is just the layout (made with CSS).

How to check the resource usage in Google Chrome: https://linuxhint.com/built_in_task_manager_google_chrome/

Alwaysin commented:
@math-GH this is the exact tool I've used to check memory status.
When using Nord theme, as soon as I scroll down, it loads every article and memory usage skyrockets. It doesn't do it on other themes.

math-GH commented:
Loading next articles while scrolling: it is a default behavior and does not depending on the theme: 
![grafik](https://user-images.githubusercontent.com/1645099/179356217-333fc92c-0379-4b31-be7c-d7cc59afd659.png)

Alwaysin commented:
Maybe I'll do a video of the behaviour if I find time to do so.
I indeed have this feature on, but on other themes, it works as described: it loads at the bottom of te page.
Ths theme, as soon as I scroll, even a millimeter, it loads everything.

math-GH commented:
OK, now I understood and can confirm that behavior.

I created an issue ticket: #4447
